lib/common/schemaUtilsCommon.js

```diff
@@ -2,7 +2,207 @@
+    'number': (d) => !isNaN(d),
+    'integer': (d) => !isNaN(d) && Number.isInteger(Number(d)),
+    'boolean': (d) => _.isBoolean(d) || d === 'true' || d === 'false',
+    'array': (d) => Array.isArray(d),
+    'object': (d) => typeof d === 'object' && !Array.isArray(d)
+  };
+
+/**
+ *  Remove the # character from the beginning of a schema path
+ * @param {string} schemaPath - a defined schemaPath
+ * @returns {string} - the schema path with # removed
+ */
+function removeSharpAndSlashFromFirstPosition(schemaPath) {
+  return schemaPath[0] === '#' ? schemaPath.slice(2) : schemaPath;
+}
+
+/**
+ *  Remove the defined word from the last position of a schema path
+ * @param {string} schemaPath - a defined schemaPath
+ * @param {string} word - word to remove
+ * @returns {string} - the schema path with type removed
+ */
+function removeWordFromLastPosition(schemaPath, word) {
+  let splittedDataPath = schemaPath.split('/');
+  if (splittedDataPath[splittedDataPath.length - 1] === word) {
+    splittedDataPath.splice(-1);
+  }
+  return splittedDataPath.join('/');
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ *
+ * @param {string} value - Value to check for
+ * @param {string} type - The type in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function compareType(value, type) {
+  return value === '<' + type + '>';
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ * Works in array types
+ * @param {string} value - Value to check for
+ * @param {*} types - The types in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function isTypeValueArrayCheck(value, types) {
+  return types.find((type) => {
+    return compareType(value, type);
+  }) !== undefined;
+}
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ * Works in array types
+ * @param {string} value - Value to check for
+ * @param {*} types - The types in the schemna
+ * @returns {Boolean} the value is the representation of its type
+ */
+function checkValueOnlyTypes(value, types) {
+  return Array.isArray(types) ? isTypeValueArrayCheck(value, types) : compareType(value, types);
+}
+
+/**
+ * Checks if value is postman variable or not
+ *
+ * @param {string} type - type to look for
+ * @param {string} format - format from schema
+ * @returns {Boolean} postman variable or not
+ */
+function getDefaultFromTypeAndFormat(type, format) {
+  return typesMap[type][format];
+}
+
+
+/**
+ * Checks if value is the representation of its type like:
+ * "<integer>"
+ * Works in array types
+ * @param {string} value - Value to check for
+ * @param {*} types - The types in the schema
+ * @param {*} format - format from the schema
+ * @returns {Boolean} the value is the representation of its type
+ */
+function checkValueTypesAndFormat(value, types, format) {
+  let typesNotInMapp = [],
```

Comment:
Is this variable naming typo intentional? I see it's consistent in the further usages.